clients/fides-js/src/lib/preferences.ts

```diff
@@ -54,11 +31,11 @@ async function savePreferencesApi(
   userLocationString?: string
 ) {
   if (options.apiOptions?.savePreferencesFn) {
-    debugLog(options.debug, "Calling custom save preferences fn");
-    await options.apiOptions.savePreferencesFn(
+    await customSaveConsentPreferences(
```

Comment:
actually yeah—this is the pattern I mean in my earlier comment, where `savePreferencesApi` is conditionally calling the custom func, and if it's not there, it continues doing whatever it originally did

Reply:
Yeah, I was back and forth on this. It's a bit tricky since `options.apiOptions.savePreferencesFn()` returns `void`, so this means if that custom fn were successful, we would still try to call the Fides base fn. We need to disable calling the Fides base fn when a custom fn is defined. I'm trying to think of a better way to track whether the fn was defined.

E.g. having `customGetPrivacyExperience` return a `boolean` (to track whether or not the fn was called) or `experience` is problematic because then we have to do an extra check everywhere we want to use the `experience` outside of the fn call to see if it's an actual experience vs `false`.

Let me know if you want me to walk you through a screenshare, as it's tough to fully describe via text 😅